packages/sanity/src/desk/comments/src/hooks/useCommentsEnabled.ts

```diff
@@ -0,0 +1,68 @@
+import {useMemo} from 'react'
+import {getPublishedId, useFeatureEnabled, useSource} from 'sanity'
+
+interface Disabled {
+  isEnabled: false
+  reason: 'disabled-by-config' | 'plan-upgrade-required' | 'loading'
+}
+
+interface Enabled {
+  isEnabled: true
+  reason: null
+}
+
+interface Loading {
+  isEnabled: false
+  reason: 'loading'
+}
+
+type CommentsEnabled = Enabled | Disabled | Loading
+
+interface CommentsEnabledHookOptions {
+  documentId: string
+  documentType: string
+}
+
+/**
+ * @internal
+ */
+export function useCommentsEnabled(opts: CommentsEnabledHookOptions): CommentsEnabled {
+  const {documentId, documentType} = opts
+
+  // 1. Plan check
+  const {enabled: featureEnabled, isLoading} = useFeatureEnabled('studioComments')
+
+  // 2. Config check
+  const {enabled} = useSource().document.unstable_comments
+  const enabledFromConfig = enabled({documentType, documentId: getPublishedId(documentId)})
+
+  const commentsEnabled = useMemo(() => {
+    if (isLoading) {
+      return {
+        isEnabled: false,
+        reason: 'loading',
+      } satisfies CommentsEnabled
```

Comment:
is there any benefit from using satisfies vs typing the return value of this function as `CommentsEnabled`? (e.g. changing from `useMemo(() => {…` to `useMemo((): CommentsEnabled => {`  above)

Reply:
No, there's no benefit! I'll change to `(): CommentsEnabled` instead of `satisfies CommentsEnabled` since that's a much cleaner in this situation 👍